src/js/common.ml

```diff
@@ -157,25 +157,116 @@ let js_parse_motoko s =
     end)
     in Js.some (js_of_sexpr (Arrange.prog prog)))
 
-let js_parse_motoko_typed paths =
-  let paths = paths |> Js.to_array |> Array.to_list in
-  let load_result = Mo_types.Cons.session (fun _ ->
-    Pipeline.load_progs Pipeline.parse_file (paths |> List.map Js.to_string) Pipeline.initial_stat_env)
+let js_parse_motoko_with_deps path s =
+  let main_file = Js.to_string path in
+  let s = Js.to_string s in
+  let prog_and_deps_result =
+    let open Diag.Syntax in
+    let* prog, _ = Pipeline.parse_string main_file s in
+    let* deps =
+      Pipeline.ResolveImport.resolve (Pipeline.resolve_flags ()) prog main_file
+    in
+    Diag.return (prog, deps)
   in
-  js_result load_result (fun (libs, progs, senv) ->
-  progs |> List.map (fun prog ->
+  js_result prog_and_deps_result (fun (prog, deps) ->
     let open Mo_def in
-    let module Arrange_sources_types = Arrange.Make (struct
+    let module Arrange = Arrange.Make (struct
       let include_sources = true
-      let include_types = true
+      let include_types = false
       let include_docs = Some prog.note.Syntax.trivia
       let include_parenthetical = false
-      let main_file = Some prog.at.left.file
-    end)
-    in object%js
-      val ast = js_of_sexpr (Arrange_sources_types.prog prog)
-      (* val typ = js_of_sexpr (Arrange_sources_types.typ typ) *)
-    end) |> Array.of_list |> Js.array |> Js.some)
+      let main_file = Some main_file
+    end) in
+    Js.some (
+      object%js
+        val ast = js_of_sexpr (Arrange.prog prog)
+        val immediateImports =
+          deps
+          |> List.map (fun dep -> Js.string (Pipeline.resolved_import_name dep))
+          |> Array.of_list
+          |> Js.array
+      end))
+
+module Map_conversion (Map : Map.S) = struct
+  let from_js
+    (type data)
+    (map : _ Js.t)
+    (from_key : 'k Js.t -> Map.key)
+    (from_data : 'd Js.t -> data)
+    : data Map.t
+  =
+  let result = ref Map.empty in
+  let callback =
+    (* [forEach] in JS gives value, key, and map, in this order. *)
+    Js.wrap_callback (fun v k _m ->
+      let k = from_key k in
+      let v = from_data v in
+      result := Map.add k v !result)
+  in
+  ignore (Js.Unsafe.meth_call map "forEach" [|Js.Unsafe.inject callback|]);
+  !result
```

Comment:
indent?

Reply:
Can you give an idea how it should look like? I have tried to run `ocamlformat` and it changes it to

```ocaml
module Map_conversion (Map : Map.S) = struct
  let from_js (type data) (map : _ Js.t) (from_key : 'k Js.t -> Map.key)
      (from_data : 'd Js.t -> data) : data Map.t =
```
(the rest is the same)

Reply:
Oh, I just preferred the body of the function to be indented (lines 198-207), but don't worry about it.

Reply:
Right, I didn't notice it

Reply:
done